From 8c1767a02f2a08127d2b66d968be486175c21367 Mon Sep 17 00:00:00 2001
From: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
Date: Fri, 27 Sep 2024 18:57:43 +0200
Subject: [PATCH 78/98] usb: xhci: move Etron PCI IDs to pci_ids.h

The same PCI IDs are needed for the new xhci-pci-etron sub-driver.
Move them to pci_ids.h so they can be re-used instead of having to
copy them over.

Signed-off-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>

[Rebased for kernel 6.12.96: on this kernel version the two device ID
macros were still named PCI_DEVICE_ID_EJ168/PCI_DEVICE_ID_EJ188 (no
_ETRON infix; that renaming happened separately upstream), so the
xhci-pci.c hunk removes those names instead and the two use sites
further down in the same function were updated to the new
PCI_DEVICE_ID_ETRON_EJ168/EJ188 names now provided by pci_ids.h.]
---
 drivers/usb/host/xhci-pci.c | 8 ++++----
 include/linux/pci_ids.h     | 4 ++++
 2 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/drivers/usb/host/xhci-pci.c b/drivers/usb/host/xhci-pci.c
--- a/drivers/usb/host/xhci-pci.c
+++ b/drivers/usb/host/xhci-pci.c
@@ -34,10 +34,6 @@
 #define PCI_DEVICE_ID_FRESCO_LOGIC_FL1100	0x1100
 #define PCI_DEVICE_ID_FRESCO_LOGIC_FL1400	0x1400

-#define PCI_VENDOR_ID_ETRON		0x1b6f
-#define PCI_DEVICE_ID_EJ168		0x7023
-#define PCI_DEVICE_ID_EJ188		0x7052
-
 #define PCI_DEVICE_ID_VIA_VL805			0x3483

 #define PCI_DEVICE_ID_INTEL_LYNXPOINT_XHCI		0x8c31
@@ -418,8 +414,8 @@
 		xhci->quirks |= XHCI_DEFAULT_PM_RUNTIME_ALLOW;

 	if (pdev->vendor == PCI_VENDOR_ID_ETRON &&
-	    (pdev->device == PCI_DEVICE_ID_EJ168 ||
-	     pdev->device == PCI_DEVICE_ID_EJ188)) {
+	    (pdev->device == PCI_DEVICE_ID_ETRON_EJ168 ||
+	     pdev->device == PCI_DEVICE_ID_ETRON_EJ188)) {
 		xhci->quirks |= XHCI_ETRON_HOST;
 		xhci->quirks |= XHCI_RESET_ON_RESUME;
 		xhci->quirks |= XHCI_BROKEN_STREAMS;
diff --git a/include/linux/pci_ids.h b/include/linux/pci_ids.h
--- a/include/linux/pci_ids.h
+++ b/include/linux/pci_ids.h
@@ -2595,6 +2595,10 @@
 
 #define PCI_VENDOR_ID_REDHAT		0x1b36
 
+#define PCI_VENDOR_ID_ETRON		0x1b6f
+#define PCI_DEVICE_ID_ETRON_EJ168	0x7023
+#define PCI_DEVICE_ID_ETRON_EJ188	0x7052
+
 #define PCI_VENDOR_ID_WCHIC		0x1c00
 #define PCI_DEVICE_ID_WCHIC_CH382_0S1P	0x3050
 #define PCI_DEVICE_ID_WCHIC_CH382_2S1P	0x3250

